pkg/util/k8sutil/self_hosted.go

```diff
@@ -32,59 +33,32 @@ const (
 	etcdLockPath               = "/var/lock/etcd.lock"
 )
 
-var (
-	envPodIP = v1.EnvVar{
-		Name: "MY_POD_IP",
-		ValueFrom: &v1.EnvVarSource{
-			FieldRef: &v1.ObjectFieldSelector{
-				APIVersion: "v1",
-				FieldPath:  "status.podIP",
-			},
-		},
-	}
-)
-
 func selfHostedDataDir(ns, name string) string {
 	return path.Join(etcdVolumeMountDir, ns+"-"+name)
 }
 
-func PodWithAddMemberInitContainer(p *v1.Pod, endpoints []string, ns, name, peerURL string, cs spec.ClusterSpec) *v1.Pod {
```

Comment:
If we try to add member in init container, it would fail because it couldn't resolve the host name. The host name would not exist if pod in "Initializing" state, but exist in "Running".